sdk/eventhub/azure-eventhub-checkpointstoretable/tests/test_storage_table_partition_manager.py

```diff
@@ -1,8 +1,196 @@
+    with pytest.raises(OwnershipLostError) as e_info:
+            checkpoint_store.claim_ownership(ownership)
+
+def _claim_and_list_ownership(storage_connection_str, table_name):
+    fully_qualified_namespace = 'test_namespace'
+    eventhub_name = 'eventhub'
+    consumer_group = '$default'
+    ownership_cnt = 8
+
+    checkpoint_store = TableCheckpointStore.from_connection_string(
+        storage_connection_str, table_name)
+    ownership_list = checkpoint_store.list_ownership(
+            fully_qualified_namespace,
+            eventhub_name,
+            consumer_group)
+    assert len(ownership_list) == 0
+
+    ownership_list = []
+
+    for i in range(ownership_cnt):
+            ownership = {}
+            ownership['fully_qualified_namespace'] = fully_qualified_namespace
+            ownership['eventhub_name'] = eventhub_name
+            ownership['consumer_group'] = consumer_group
+            ownership['owner_id'] = 'owner_id'
+            ownership['partition_id'] = str(i)
+            ownership['etag'] = None
+            ownership['last_modified_time'] = None
+            ownership_list.append(ownership)
+    newownershiplist = checkpoint_store.claim_ownership(ownership_list)
+    assert ownership_list != newownershiplist
+    assert len(newownershiplist) == len(ownership_list)
+    for i in range(len(ownership_list)):
+        assert ownership_list[i]['etag'] != newownershiplist[i]['etag']
+        assert ownership_list[i]['last_modified_time'] != newownershiplist[i]['last_modified_time']
+
+    ownership_list = checkpoint_store.list_ownership(fully_qualified_namespace,eventhub_name,consumer_group)
+    assert len(ownership_list) == ownership_cnt
+    assert len(ownership_list) == len(newownershiplist)
+    for i in range(len(newownershiplist)):
+        assert ownership_list[i]['etag'] == newownershiplist[i]['etag']
+        assert ownership_list[i]['last_modified_time'] == newownershiplist[i]['last_modified_time']
+
+def _update_and_list_checkpoint(storage_connection_str, table_name):
+    fully_qualified_namespace = 'test_namespace'
+    eventhub_name = 'eventhub'
+    consumer_group = '$default'
+    partition_cnt = 8
+
+    checkpoint_store = TableCheckpointStore.from_connection_string(
+        storage_connection_str, table_name)
+    checkpoint_list = checkpoint_store.list_checkpoints(
+            fully_qualified_namespace,
+            eventhub_name,
+            consumer_group)
+    assert len(checkpoint_list) == 0
+    for i in range(partition_cnt):
+            checkpoint = {
+                'fully_qualified_namespace': fully_qualified_namespace,
+                'eventhub_name': eventhub_name,
+                'consumer_group': consumer_group,
+                'partition_id': str(i),
+                'offset': 2,
+                'sequence_number': 20
+            }
```

Comment:
can also move this to helper function if used multiple times

Reply:
okay will do